happi/audit.py

```diff
@@ -0,0 +1,424 @@
+
+    def process_args(self, database_path, args):
+        """
+        Checks to see if a valid path to database was provided.
+        If --extras is provided, will call for check xtra attributes, otherwise
+        it will just proceed with parsing and validating the database call.
+        """
+        if self.validate_file(database_path):
+            logger.info('Using database file at %s ', database_path)
+            if args.extras:
+                self.check_extra_attributes(database_path)
+            else:
+                self.parse_database(database_path)
+        else:
+            logger.error('The database %s file path '
+                         'could not be validated', database_path)
+            sys.exit(1)
+
+    def validate_file(self, file_path):
+        """
+        Tests whether a path is a regular file
+
+        Parameters
+        -----------
+        file_path: str
+            File path to be validate
+
+        Returns
+        -------
+        bool
+            To indicate whether the file is a valid regular file
+        """
+        return os.path.isfile(file_path)
+
+    def validate_container(self, item):
+        """
+        Validates container definition
+        """
+        container = item.get('type')
+        device = item.get('name')
+        if container and container not in registry:
+            logger.error('Invalid device container: %s for device %s',
+                         container, device)
+            return self.report_code.INVALID
+        elif not container:
+            logger.error('No container provided for %s', device)
+            return self.report_code.MISSING
+        else:
+            # seems like the container has been validated
+            return self.report_code.SUCCESS
+
+    def validate_args(self, item):
+        """
+        Validates the args of an item
+        """
+        return [self.create_arg(item, arg) for arg in item.args]
+
+    def validate_kwargs(self, item):
+        """
+        Validates the kwargs of an item
+        """
+        return dict((key, self.create_arg(item, val))
+                    for key, val in item.kwargs.items())
+
+    def create_arg(self, item, arg):
+        """
+        Function borrowed from loader to create
+        correctly typed arguments from happi information
+        """
+        if not isinstance(arg, str):
+            return arg
+        return fill_template(arg, item, enforce_type=True)
+
+    def get_device_class(self, item):
+        """
+        Validates device_class field
+        """
+        device_class = item.get('device_class')
+        device = item.get('name')
+
+        if not device_class:
+            logger.warning('Detected a None vlaue for %s. '
+                           'The device_class cannot be None', device)
+            return self.report_code.MISSING
+        else:
+            try:
+                mod, cls = device_class.rsplit('.', 1)
+            except (Exception, ValueError) as e:
```

Comment:
```suggestion
            except Exception as e:
```
Exception will catch pretty much everything including `ValueError`